src/script/lua_api/l_internal.h

```diff
@@ -43,3 +43,22 @@ with this program; if not, write to the Free Software Foundation, Inc.,
 #define GET_ENV_PTR         \
 	MAP_LOCK_REQUIRED;      \
 	GET_ENV_PTR_NO_MAP_LOCK
+
+
+/**
+ * This macro permits to bootstrap easily a Lua Referenced object
+ */
+#define LUAREF_OBJECT(name)                                                                                        \
+private:                                                                                                           \
+    name *m_object = nullptr;                                                                                                \
+    static const char className[];                                                                                 \
+    static const luaL_Reg methods[];                                                                               \
+    static int gc_object(lua_State *L);                                                                            \
+                                                                                                                   \
+public:                                                                                                            \
+    explicit Lua##name(name *object);                                                                           \
+    virtual ~Lua##name();                                                                                       \
+    static void Register(lua_State *L);                                                                            \
+    static void create(lua_State *L, name *object);                                                                \
+    static Lua##name *checkobject(lua_State *L, int narg);                                                      \
```

Comment:
Minor issue: Backslash placement is outside of our 80 char limit and not on the same level for the entire `define`.